pkg/cassandra/config/config.go

```diff
@@ -134,8 +177,32 @@ type SessionBuilder interface {
 	NewSession() (cassandra.Session, error)
 }
 
+func (c *Configuration) newSessionPrerequisites() error {
+	cluster, err := c.NewCluster()
+	if err != nil {
+		return err
+	}
+
+	cluster.Keyspace = ""
+
+	session, err := cluster.CreateSession()
+	if err != nil {
+		return err
+	}
+
+	wSession := gocqlw.WrapCQLSession(session)
+	defer wSession.Close()
+
+	return GenerateSchemaIfNotPresent(wSession, &c.Schema)
+}
+
 // NewSession creates a new Cassandra session
 func (c *Configuration) NewSession() (cassandra.Session, error) {
+	err := c.newSessionPrerequisites()
```

Comment:
The issue lies here: [func (c *Configuration) NewCluster() (*gocql.ClusterConfig, error) {](https://github.com/jaegertracing/jaeger/blob/295146c8d419a3cf4ced9a178826c337f9eb440e/pkg/cassandra/config/config.go#L151-L153)

This uses: `cluster.Keyspace = c.Schema.Keyspace` for creating a connection. What if `c.Schema.Keyspcae` doesn't exist? -> The connection will not be established.

That's why in `newSessionPrerequisites`, I am creating a session without `cluster.Keyspace`, and creating `c.Schema.Keyspace` if it doesn't exist using `GenerateSchemaIfNotPresent`. After this, a new session will be established with `cluster.Keyspace = c.Schema.Keyspace`, so that we directly read/write from/to this keyspace.

I thought of something like creating a session without keyspace and then updating the session to use that keyspace, but it looks like it is not possible: https://pkg.go.dev/github.com/gocql/gocql#readme-important-default-keyspace-changes